### PR TITLE
[DE-869] Exporting current content design

### PR DESCRIPTION
**Pull request by Cromeror**
Before was exported the content of campaign editor and now is export the design that is current edited on the unlayer editor.

### CODE REVIEW

File: src/components/Campaign.tsx [hunk 1]
```diff
@@ -68,17 +78,23 @@ export const Campaign = () => {
                   <li>
                     <button
                       type="button"
-                      onClick={() => setOpen(true)}
-                      className="dp-button button-medium primary-green"
+                      onClick={() => exportContent()}
```
Comment (andresmoschini) [inline src/components/Campaign.tsx:81]: Hay una demora entre que se hace click en el botón y cuando se abre el modal, correcto?

Que ocurre si se hace click varias veces? se exporta varias veces?

File: src/components/SingletonEditor.tsx [hunk 2]
```diff
@@ -102,6 +104,48 @@ export const useSingletonEditor = (
...
+    );
+
+    const [htmlExport, imageExport] = await Promise.all([
+      exportHtml(),
+      exportImage(),
+    ]);
+
+    if (!htmlExport.design) {
+      //TODO: implement a better solution when occurs this error
+      console.error("The model exported don´t contain design");
```
Comment (andresmoschini) [inline src/components/SingletonEditor.tsx:133]: Very minor:

```suggestion
      // TODO: implement a better solution when occurs this error
      console.error("The model exported don't contain design");
```

File: src/components/SingletonEditor.tsx [hunk 3]
```diff
@@ -102,6 +104,48 @@ export const useSingletonEditor = (
...
+    if (!editorState.isLoaded) {
+      console.error("The editor is loading, can't save yet!");
+      return;
+    }
+    if (!onExport) {
+      console.error("The onExport callback must be configured");
+      return;
+    }
+
+    setIsExporting(true);
```
Comment (andresmoschini) [inline src/components/SingletonEditor.tsx:117]: It should be set to `false` again in case of an error, right?

Comment (andresmoschini) [reply src/components/SingletonEditor.tsx:117]: I am talking about [line #131](https://github.com/FromDoppler/doppler-editors-webapp/pull/412/files#diff-5886b9902d1e2302004c7c6185c63c9950e6aa9cb7a3739434720fc70be3eee5R131) `if (!htmlExport.design) {`

File: src/components/Campaign.tsx [hunk 4]
```diff
@@ -27,12 +28,16 @@ export const Campaign = () => {
...
       onSave: (content: Content) => {
         campaignContentMutation.mutate({ idCampaign, content });
       },
+      onExport: (content: UnlayerContent) => {
+        setContentToExport(content);
+        setOpen(true);
+      },
     },
     [campaignContentQuery.data, campaignContentMutation.mutate, idCampaign]
   );
```
Comment (andresmoschini) [inline src/components/Campaign.tsx:43]:
```suggestion
  const { save, export, isExporting } = useSingletonEditor(
    {
      initialContent: campaignContentQuery.data,
      onSave: (content: Content) => {
        campaignContentMutation.mutate({ idCampaign, content });
      },
    },
    [campaignContentQuery.data, campaignContentMutation.mutate, idCampaign]
  );

  const exportContent = (content: UnlayerContent) => {
      setContentToExport(content);
      setOpen(true);
  };
```

I am not totally sure, but in my opinion, it looks cleaner and it is more flexible.

Comment (andresmoschini) [reply src/components/Campaign.tsx:43]: `save` is different because it is used internally by the `SingletonEditor`.

Comment (andresmoschini) [reply src/components/Campaign.tsx:43]: What is more! you do not need `isExporting`: 

```suggestion
  const [ isExportingAstemplate, setIsExportingAstemplate ] = useState(false);

  const { save, export } = useSingletonEditor(
    {
      initialContent: campaignContentQuery.data,
      onSave: (content: Content) => {
        campaignContentMutation.mutate({ idCampaign, content });
      },
    },
    [campaignContentQuery.data, campaignContentMutation.mutate, idCampaign]
  );

  const exportAsTemplate = () => {
    setIsExportingAsTemplate(true);
    try {
      const content = await export();
      setContentToExportAsTemplate(content);
      setExportAsTemplateDialogoOpen(true);
    } finally {
      setIsExportingAsTemplate(false);
    }
  };
```

File: src/components/SingletonEditor.tsx (hunk 3 above)
Comment (Cromeror) [reply src/components/SingletonEditor.tsx]: This was move to campaing implementation.

File: src/components/Campaign.tsx (hunk 1 above)
Comment (Cromeror) [reply src/components/Campaign.tsx]: Antes se ejecutaba el export varias veces, ahora se deshabilita el botón hasta que finalice de exportar.

File: src/components/Campaign.tsx (hunk 4 above)
Comment (Cromeror) [reply src/components/Campaign.tsx]: Good approach, was implemented. 👍🏻

File: src/components/Campaign.tsx [hunk 5]
```diff
@@ -19,15 +19,17 @@ export const errorMessageTestId = "error-message";
 export const editorTopBarTestId = "editor-top-bar-message";
 
 export const Campaign = () => {
-  const [isOpen, setOpen] = useState(false);
+  const [contentToExport, setContentToExport] = useState<UnlayerContent>();
+  const [isOpen, toggleModal] = useState(false);
+  const [isExporting, setExportingState] = useState(false);
```
Comment (andresmoschini) [inline src/components/Campaign.tsx:24]: I think that these names would be better.

```suggestion
  const [contentToExportAsTemplate, setContentToExportAsTemplate] = useState<UnlayerContent>();
  const [isExportAsTemplateModalOpen, setIsExportAsTemplateModalOpen] = useState(false);
  const [isExportingAsTemplate, setIsExportingAsTemplate] = useState(false);
```

An extra note about _toggle_. If I understand it well, "toggle" is not the right word when you specify the next state.

An extra note about _setExportingState_. I think that using the convention of prepending `set` to the _state name_ (`[ stateName, setStateName ] = useState()`) is simpler.

File: src/components/Campaign.tsx [hunk 6]
```diff
@@ -48,6 +50,19 @@ export const Campaign = () => {
...
+    toggleModal(false);
+    setContentToExport(undefined);
+  };
+
+  const openExportModal = async () => {
+    setExportingState(true);
+    const content = await exportContent();
+    setContentToExport(content);
+    toggleModal(true);
+    setExportingState(false);
```
Comment (andresmoschini) [inline src/components/Campaign.tsx:63]:
```suggestion
    try {
      const content = await exportContent();
      setContentToExport(content);
    } finally {
      toggleModal(true);
      setExportingState(false);
    }
```

**Pull request merged**